docs/guide/tutorials/task-guide.md

```diff
@@ -0,0 +1,524 @@
+
+@task(name="A")
+def task_A():
+    # does something interesting and stateful
+    return None
+
+
+@task(name="B")
+def task_B():
+    # also does something interesting and stateful
+    return None
+
+with Flow("functional-example") as flow:
+    result = task_B(upstream_tasks=[task_A])
+```
+:::
+
+## Mapping
+
+[Mapping](https://docs.prefect.io/guide/core_concepts/mapping.html) is a unique feature of Prefect, which allows users to _dynamically_ spawn multiple copies of a given Task in response to an upstream Task's output.  When a task is mapped, Prefect automatically creates a copy of the task for each element of its input data. The copy -- referred to as a "child" task -- is applied only to that element. This means that mapped tasks actually represent the computations of many individual children tasks.
+
+If a "normal" (non-mapped) task depends on a mapped task, Prefect automatically applies a reduce operation to gather the mapped results and pass them to the downstream task.
+
+However, if a mapped task relies on another mapped task, Prefect does not reduce the upstream result. Instead, it connects the nth upstream child to the nth downstream child, creating independent parallel pipelines.
+
+Let's have a look at a simple example:
+
+```python
+import prefect
+from prefect import task, Flow
+
+@task
+def add_one(x):
+    """Operates one number at a time"""
+    return x + 1
+
+@task(name="sum")
+def sum_numbers(y):
+    """Operates on an iterable of numbers"""
+    result = sum(y)
+    logger = prefect.context["logger"]
+    logger.info("The sum is {}".format(result))
+    return result
+
+
+## Imperative API
+flow = Flow("Map Reduce")
+flow.set_dependencies(add_one, keyword_tasks={"x": [1, 2, 3, 4]}, mapped=True)
+flow.set_dependencies(sum_numbers, keyword_tasks={"y": add_one})
+
+## Functional API
+with Flow("Map Reduce") as flow:
+    mapped_result = add_one.map(x=[1, 2, 3, 4])
+    summed_result = sum_numbers(mapped_result)
+```
+
+Whenever this Flow is run, we will see the following log:
+```
+[2019-07-20 21:35:00,968] - INFO - prefect.Task | The sum is 14
+```
+
+Note that each "child" task is a _first class Prefect Task_.  This means that they can do anything a "normal" task can do, including succeed, fail, retry, pause, or skip.
+
+#### Not Everything must be mapped over
+
+Suppose we have a Task that accepts many keyword arguments and we want to only map over a _subset_ of those arguments.  In this case, we can use Prefect's `unmapped` container for specifying those inputs which should _not_ be mapped over:
+
+```python
+from prefect import task, Flow
+from prefect.utilities.tasks import unmapped
+
+@task
+def add_one(x, y):
+    """Operates one number at a time"""
+    return x + y
+
+with Flow("unmapped example") as flow:
+    result = add_one.map(x=[1, 2, 3], y=unmapped(5))
+```
+When this Flow runs, only the `x` keyword will be mapped over; the `y` will remain fixed with the constant value of 5.
```

Comment:
Can you add the output of this if run? And also maybe note what happens if `y=5` (not unmapped)

Reply:
Yea I can; there are just a lot of logs.  I just won't include all the other logs though.

Reply:
I didn't update this because there were so many logs and I honestly wasn't sure which logs to choose.